builder/builder.go

```diff
@@ -642,3 +640,37 @@ func validateBuildkitEndpoint(ep string) (string, error) {
 	}
 	return ep, nil
 }
+
+// parseBuildkitFlags parses buildkit flags
+func parseBuildkitFlags(inp string, driver string) (res []string, err error) {
+	if inp != "" {
+		res, err = shlex.Split(inp)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse buildkit flags")
+		}
+	}
+
+	if driver == "kubernetes" || driver == "docker-container" {
+		var allowInsecureEntitlements []string
+		flags := pflag.NewFlagSet("buildkitd", pflag.ContinueOnError)
+		flags.Usage = func() {}
+		flags.StringArrayVar(&allowInsecureEntitlements, "allow-insecure-entitlement", nil, "")
+		_ = flags.Parse(res)
+
+		var hasNetworkHostEntitlement bool
+		for _, e := range allowInsecureEntitlements {
+			if e == "network.host" {
+				hasNetworkHostEntitlement = true
+				break
+			}
+		}
+
+		if !hasNetworkHostEntitlement {
```

Comment:
I think this condition should be "if no insecure entitlements set by user"

Reply:
you mean we should check for any insecure entitlements? So if `security.insecure` or `network.host` or whatever is set for `--allow-insecure-entitlement` then we should skip setting `--allow-insecure-entitlement=network.host`?

Reply:
yes, just to leave a way for users to manually configure entitlements as well to be as they want them, without us overwriting